### PR TITLE
Test performance of sign/verify primitives

### PR DESCRIPTION
**Pull request by beejones**
**Problem:**
Adding performance tests for the crypto primitives

**Solution:**
Add special test so Logcat can show the performance with filter: PerfTest->

**Validation:**
New unit tests

**Type of change:**
- [ ] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [x] Test
- [ ] Logging/Telemetry

**Risk**:
- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [x] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: sdk/src/androidTest/java/com/microsoft/did/sdk/crypto/protectors/TokenSignerPerformanceTest.kt [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        when (result) {
+            is Result.Success -> {
+                // use result.payload (Identifier)
+                val signedPayload = signer.signWithIdentifier(payload, result.payload)
+                println("PerfTest->(${getTestName()}) - End sign: ${timer(startTime)}")
+                println("PerfTest->(${getTestName()}) - Signed token: ${signedPayload}")
+            }
+        }
+
+        println("PerfTest->(${getTestName()}) - Start verify")
```
Comment (nithyaganeshng) [inline sdk/src/androidTest/java/com/microsoft/did/sdk/crypto/protectors/TokenSignerPerformanceTest.kt:97]: TokenSigner class does only signing I believe. Can we remove print statements for verification here?

File: sdk/src/androidTest/java/com/microsoft/did/sdk/crypto/protectors/TokenSignerPerformanceTest.kt [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+                    "KeyReference" to keyReference
+                )
+            ), true, listOf(KeyUsage.Sign)
+        )
+
+        cryptoOperations = CryptoOperations(ellipticCurveSubtleCrypto, keyStore, EllipticCurvePairwiseKey())
+        val sidetreePayloadProcessor = SidetreePayloadProcessor(defaultTestSerializer)
+        identifierCreator = IdentifierCreator(cryptoOperations, sidetreePayloadProcessor)
+
+        signer = TokenSigner(cryptoOperations, serializer)
```
Comment (gproanomsft) [inline sdk/src/androidTest/java/com/microsoft/did/sdk/crypto/protectors/TokenSignerPerformanceTest.kt:64]: I don't know if authenticator uses ahead of time or just in time compilation, but generally running once makes for  an inaccurate perf assessment, in the case of JIT - you spend a lot of time not executing your code

absent a dedicated benchmarking framework, in the past I would benchmark by running the code once to ensure any JIT occurs, and then running in a 10k loop

File: sdk/src/main/java/com/microsoft/did/sdk/crypto/keyStore/AndroidKeyStore.kt [hunk 3]
```diff
@@ -57,53 +65,88 @@ class AndroidKeyStore @Inject constructor(private val context: Context, private
...
-        val nativeKeys = listNativeKeys()
-        var key = nativeKeys[keyReference]
-        if (key != null) {
-            return KeyContainer(
-                kty = key.kty,
-                keys = key.kids.map {
-                    AndroidKeyConverter.androidPrivateKeyToPrivateKey(it, keyStore)
-                }
-            )
+        var privateKey = nativePrivateKeyCache.get(keyReference)
```
Comment (dangodb) [inline sdk/src/main/java/com/microsoft/did/sdk/crypto/keyStore/AndroidKeyStore.kt:68]: Are we using both the native and software keystores at this point? if not, let's get rid of one of them. We can reintroduce it later when we need it

File: sdk/src/main/java/com/microsoft/did/sdk/crypto/keyStore/AndroidKeyStore.kt [hunk 4]
```diff
@@ -35,6 +35,14 @@ class AndroidKeyStore @Inject constructor(private val context: Context, private
 
     companion object {
         const val provider = "AndroidKeyStore"
+        // caches
+        private val nativePrivateKeyCache = HashMap<String, KeyContainer<PrivateKey>>()
```
Comment (dangodb) [inline sdk/src/main/java/com/microsoft/did/sdk/crypto/keyStore/AndroidKeyStore.kt:39]: How long does the AndroidKeyStore lives for in memory? Are we concerned about keeping the secrets in clear in memory for that long? Would it be better to only warmup the required secrets and dispose of them when the signature is done?

File: sdk/src/main/java/com/microsoft/did/sdk/crypto/keyStore/AndroidKeyStore.kt [hunk 5]
```diff
@@ -231,10 +277,15 @@ class AndroidKeyStore @Inject constructor(private val context: Context, private
                 }
             }
         }
+        nativeList = output
         return output
     }
 
     private fun listSecureData(): Map<String, KeyStoreListItem> {
+        if (secureDataList != null) {
```
Comment (dangodb) [inline sdk/src/main/java/com/microsoft/did/sdk/crypto/keyStore/AndroidKeyStore.kt:285]: Does this has a big perf impact? IIRC sharedPrefs does a load from disk the 1st time and then use the inmemory cache for subsequent fetch. Does EncryptedSharedPreferences has a similar concept? If so, this guy may not be needed. I'm a bit concerned about the number of caches that we have in this class.

**Review by dangodb — CHANGES_REQUESTED** [review]

**Comment by AndiMiko** [comment]
Obsolete due to crypto rework

**Pull request closed without merging**